File: easy/0674_longest_continuous_increasing_subsequence/solution.cpp
#include <vector>

class Solution {
public:
  int findLengthOfLCIS(std::vector<int> &nums) {
    const size_t n = nums.size();

    int maxLen = 1;

    for (size_t i = 1; i < n; ++i) {
      int currLen = 1;

      while (i < n && nums[i - 1] < nums[i]) {
        ++currLen;
        ++i;
      }

      maxLen = std::max(maxLen, currLen);
    }

    return maxLen;
  }
};

File: easy/0674_longest_continuous_increasing_subsequence/solution.py
from typing import List


class Solution:
    def findLengthOfLCIS(self, nums: List[int]) -> int:
        n = len(nums)

        maxLen = 1

        i = 1
        while i < n:
            currLen = 1

            while i < n and nums[i - 1] < nums[i]:
                currLen += 1
                i += 1

            maxLen = max(maxLen, currLen)
            i += 1

        return maxLen
